macros/set_warehouse_size.sql

```diff
@@ -8,23 +8,24 @@
 
 {% macro snowflake__set_warehouse_size(size) %}
 
-	{% if var("available_sizes", None) == None %}
-		{{ exceptions.raise_compiler_error("Please set the `available_sizes` variable in the dbt_project.yml.") }}
+	{% if var("snowflake_warehouses", None) == None %}
+		{{ exceptions.raise_compiler_error("Please set the `snowflake_warehouses` variable in the dbt_project.yml.") }}
 	{% endif %}
 
-    {% if size not in var("available_sizes") %}
-        {{ exceptions.raise_compiler_error("Warehouse size not one of " ~ var("available_sizes")) }}
-    {% endif %}
-
-    {% if target.name == 'prod' %}
-        {{ return('PROD_DBT_WH_' ~ size) }}
-    {% elif target.name == 'dev' %}
-        {{ return('DEV_DBT_WH_' ~ size) }}
-    {% elif target.name == 'ci' %}
-        {{ return('CI_WH_' ~ size) }}
+    {% set size_dict = var('snowflake_warehouses') %}
+    {% set target_name = target.name %}
+    
+    {% if target.name in size_dict %}
+        {% set env = size_dict[target_name] %}
+        {% if env['size']|length == 0 %}
+            {{ env['warehouse_prefix'] }}
```

Comment:
B: So if there are no sizes specified - we default to `warehouse_prefix`? What if this is also not defined? What is the default-default behavior if the size_dict is not specified?

Reply:
Good point I updated the if statement to:
1. If `env['size']` doesnt exist > raise exception
2. If `env['size']` if empty > no sizes are specified thus we just use the warehouse prefix
3. If `size` is in `env['size']` > return warehouse prefix + warehouse size
4. Else: raise exception 

This means that no size key means 'I forgot it to specify' an empty list in the key means 'This warehouse has just 1 default size and that is not a suffix'